src/Build/Evaluation/Evaluator.cs

```diff
@@ -962,9 +969,10 @@ private void PerformDepthFirstPass(ProjectRootElement currentProjectOrImport)
             if (!Traits.Instance.EscapeHatches.IgnoreTreatAsLocalProperty)
             {
                 IList<string> globalPropertiesToTreatAsLocals = _expander.ExpandIntoStringListLeaveEscaped(currentProjectOrImport.TreatAsLocalProperty, ExpanderOptions.ExpandProperties, currentProjectOrImport.TreatAsLocalPropertyLocation);
-
-                foreach (string propertyName in globalPropertiesToTreatAsLocals)
+                var globalPropertiesToTreatAsLocalsCount = globalPropertiesToTreatAsLocals.Count;
```

Comment:
Add a comment here to avoid someone changing this to a foreach - is this to avoid the boxing the enumerator?